### PR TITLE
CMake integration for the master branch

### PR DESCRIPTION
**Pull request by vancraar**

### CODE REVIEW

File: distributedcombigrid/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+if(DISCOTEC_TEST)
+    enable_testing()
+    add_subdirectory(tests)
+endif()
+
+
+
+
+
+
```
Comment (freifrauvonbleifrei) [inline distributedcombigrid/CMakeLists.txt:21]: wow that is many white lines ;)

File: distributedcombigrid/src/sgpp/distributedcombigrid/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+endif ()
+
+option(DISCOTEC_USENONBLOCKINGMPICOLLECTIVE "Use non-blocking MPI collective operations" OFF)
+if (DISCOTEC_USENONBLOCKINGMPICOLLECTIVE)
+    target_compile_definitions(discotec PRIVATE USENONBLOCKINGMPICOLLECTIVE)
+endif ()
+
+#USE_NON_BLOCKING_COLLECTIVES #TODO: @polinta wird nirgends in SCONS definiert also ka wann definieren
+#ALT_LEVEL_VECTOR #TODO: @polinta wird nirgends in SCONS definiert also ka wann definieren
+#SIMFTMPI #TODO: @polinta: what is this? in confg.hpp the same is set no matter if defined or not ...
```
Comment (freifrauvonbleifrei) [inline distributedcombigrid/src/sgpp/distributedcombigrid/CMakeLists.txt:129]: yup these are all kind of outdated -- remove? I don't understand the last one either, for the others I'm pretty sure they're unused.

File: distributedcombigrid/src/sgpp/distributedcombigrid/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+#ISGENE #TODO: handle if access to GENE
+
+# Handle dependencies
+find_package(MPI REQUIRED)
+target_link_libraries(discotec PUBLIC MPI::MPI_CXX)
+
+include(FetchContent)
+
+find_package(Boost QUIET COMPONENTS serialization headers program_options unit_test_framework system ) #TODO: add all required components
```
Comment (freifrauvonbleifrei) [inline distributedcombigrid/src/sgpp/distributedcombigrid/CMakeLists.txt:138]: I've started to use boost-filesystem in some of the examples iirc

File: distributedcombigrid/src/sgpp/distributedcombigrid/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+        set(BUILD_TESTING OFF CACHE BOOL "" FORCE)
+        set(HDF5_BUILD_TOOLS OFF CACHE BOOL "" FORCE)
+        set(TEST_SHELL_SCRIPTS OFF CACHE BOOL "" FORCE)
+        set(HDF5_TEST_TOOLS OFF CACHE BOOL "" FORCE)
+        set(HDF5_TEST_EXAMPLES OFF CACHE BOOL "" FORCE)
+        set(HDF5_TEST_SWMR OFF CACHE BOOL "" FORCE)
+        set(HDF5_TESTS_PARALLEL OFF CACHE BOOL "" FORCE)
+        set(HDF5_BUILD_EXAMPLES OFF CACHE BOOL "" FORCE)
+        set(HDF5_BUILD_CPP_LIB OFF CACHE BOOL "" FORCE)
+        set(HDF5_BUILD_FORTRAN OFF CACHE BOOL "" FORCE)
```
Comment (freifrauvonbleifrei) [inline distributedcombigrid/src/sgpp/distributedcombigrid/CMakeLists.txt:234]: hm, DisCoTec uses HDF5 only through HighFive. But! If we use selalib, that uses HDF5 directly and requires the Fortran bindings.

Comment (vancraar) [reply distributedcombigrid/src/sgpp/distributedcombigrid/CMakeLists.txt]: I added a TODO, so we don't forget enabling it.

File: distributedcombigrid/src/sgpp/distributedcombigrid/CMakeLists.txt (hunk 3 above)
Comment (vancraar) [reply distributedcombigrid/src/sgpp/distributedcombigrid/CMakeLists.txt]: added

File: distributedcombigrid/CMakeLists.txt (hunk 1 above)
Comment (vancraar) [reply distributedcombigrid/CMakeLists.txt]: removed

File: distributedcombigrid/src/sgpp/distributedcombigrid/CMakeLists.txt (hunk 2 above)
Comment (vancraar) [reply distributedcombigrid/src/sgpp/distributedcombigrid/CMakeLists.txt]: removed

**Review by freifrauvonbleifrei — APPROVED** [review]
thanks so much for the new build system and clean-up!
the Jenkins tests for your branch are up and running neatly.

**Pull request merged**